src/coreComponents/codingUtilities/Section.cpp

```diff
@@ -0,0 +1,131 @@
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file Section.cpp
+ */
+
+#include "Section.hpp"
+#include <algorithm>
+
+namespace geos
+{
+
+Section::Section():
+  m_rowMinWidth( 20 )
+{}
+
+void Section::setName( string_view title )
+{
+  m_sectionTitle = title;
+}
+
+void Section::addDescription( string const & description )
+{
+  m_descriptions.push_back( description );
+}
+
+void Section::setMinWidth( integer const & minWidth )
+{
+  m_rowMinWidth = minWidth;
+}
+
+void Section::computeMaxRowSize( string_view title,
+                                 std::vector< string > const & rowsDescription )
+{
+  std::string const footerStr = "End : ";
+  integer const titleLength = footerStr.length() + title.length() + m_marginBorder * 2 + m_nbSpecialChar * 2;
+
+  m_rowLength = std::max( m_rowMinWidth, titleLength );
+
+  if( rowsDescription.size() == 0 )
+  {
+    return;
+  }
+
+  auto it = std::max_element( rowsDescription.begin(),
+                              rowsDescription.end(),
+                              []( auto const & a, auto const & b ) {
+    return a.size() < b.size();
+  } );
+
+  string const maxDescriptionSize = *it;
+
+  integer const maxDescriptionLength = integer( maxDescriptionSize.length()) + m_marginBorder * 2 + m_nbSpecialChar * 2;
+
+  m_rowLength = std::max( maxDescriptionLength, m_rowLength );
+}
+
+void Section::formatAndInsertDescriptions( std::string const & descriptionName,
+                                           std::vector< string > const & descriptionValues )
+{
+  integer const descriptionLength = descriptionName.length() + m_embeddingName;
+  m_descriptions.push_back( GEOS_FMT( "- {}: {}", descriptionName, descriptionValues[0] ) );
+  for( size_t idxValue = 1; idxValue < descriptionValues.size(); idxValue++ )
+  {
+    m_descriptions.push_back( GEOS_FMT( "{:>{}}{}", " ",
+                                        descriptionLength,
+                                        descriptionValues[idxValue] ) );
+  }
+}
+
+void Section::begin( std::ostream & oss )
+{
+  computeMaxRowSize( m_sectionTitle, m_descriptions );
+
+  string const lineSection =  GEOS_FMT( "{:#>{}}\n", "", m_rowLength );
+  integer const titleLength = m_rowLength - m_nbSpecialChar * 2;
+  integer const descriptionLength = m_rowLength - m_nbSpecialChar * 2 - m_marginBorder;
+
+  //build section title
+  oss << '\n';
+  oss << lineSection;
+  oss << GEOS_FMT( "##{:^{}}##\n", m_sectionTitle, titleLength );
+
+  //build section descriptions
+  oss << lineSection;
+
+  if( m_descriptions.empty())
+  {
+    oss << '\n';
+    return;
+  }
+
+  for( string const & description : m_descriptions )
+  {
+    oss << GEOS_FMT( "##{:<{}}{:<{}}##", " ", m_marginBorder, description, descriptionLength );
+    if( &description != &m_descriptions.back())
+    {
+      oss << '\n';
+    }
+    else
+    {
+      oss << "\n\n";
+    }
+  }
```

Comment:
I would avoid breaking the code flow with `return` at the middle:
```suggestion
  if( !m_descriptions.empty())
  {
    for( string const & description : m_descriptions )
    {
      oss << GEOS_FMT( "##{:<{}}{:<{}}##", " ", m_marginBorder, description, descriptionLength );
      if( &description == &m_descriptions.back())
      {
        oss << '\n';
      }
    }
  }
  oss << '\n';
```
Not the only place for this remark. I suggest getting some information about good practices about that.